packages/truffle-compile/index.js

```diff
@@ -35,12 +35,12 @@ var compile = function(sources, options, callback) {
 
   expect.options(options.compilers, ["solc"]);
 
-  options.compilers.solc.settings = options.compilers.solc.settings || {};
+  options.compilers.solc.settings =
+    options.compilers.solc.settings || options.compilers.solc || {};
```

Comment:
This is a bit risky, but probably fine. I think ideally we would want to grab only `optimizer` and other _specific_ keys, cause otherwise `version`, etc. will get passed as `settings`, which might cause unforeseen namespace collisions.